gulpfile.js

```diff
@@ -107,57 +101,72 @@ function processCss(inputStream, taskType) {
         .pipe(gulp.dest(paths.styles.core.dest))
         .pipe(browserSync.stream())
         .pipe($.if(flags.notify, $.notify({ message: taskType + ' task complete' })));
-} // /function processCss
+} // /function compileSASS
 
 /* -----------------------------------------------------------------------------
- * minifyJS
- * Accepts: a gulp stream, gulp-uglify options, libType, filename
+ * minifySrc
+ * Accepts: a gulp stream, a source type, gulp-uglify options, libType, filename
  * Returns: the processed stream
  * Concat stream, minify using options and write to JS src/dest using filename
  * -------------------------------------------------------------------------- */
-function minifyJS (sourceStream, uglifyOptions, libType, filename) {
+function minifySrc (sourceType, libType, filename, minifierOptions, sourceStream) {
+    function getMinifier () {
+        switch (sourceType) {
+            case 'scripts':
+                return $.uglify(minifierOptions);
+            case 'styles':
+                return $.minifyCss(minifierOptions);
+        } // /switch (sourceType)
+    } // /function getMinifier(o)
+
     return sourceStream
         .pipe($.plumber(function (error) {
             $.util.log($.util.colors.red('Error (' + error.plugin + '): ' + error.message));
             this.emit('end');
         }))
         .pipe($.concat(filename))
-        .pipe(gulp.dest(paths.scripts[libType].src))
-        // Minify with safe mangling, preserve licenses
-        .pipe($.uglify(uglifyOptions))
+        .pipe(gulp.dest(paths[sourceType][libType].src))
+        .pipe(getMinifier())
         .pipe($.rename({ suffix: '.min' }))
-        .pipe(gulp.dest(paths.scripts[libType].dest));
-} // /function minifyJS
+        .pipe(gulp.dest(paths[sourceType][libType].dest));
+} // /function minifySrc
 
 /* -----------------------------------------------------------------------------
- * getLibsGlob
- * Accepts: either a single or an array of lib collection names
+ * getVendorLibsGlob
+ * Accepts: libType - scripts, styles, etc.
+            libCol - a single or an array of lib collection names
  * Returns: a glob pattern string
  * Generates a glob pattern from the collection name of vendor libs, as defined
- * in libs.vendor
+ * in libs[libType]
  * Can also accept an array of lib collection names - e.g. ['ie8', 'core']
  * -------------------------------------------------------------------------- */
-function getLibsGlob (libCol) {
+function getVendorLibsGlob (libType, libCol) {
     // 1. Set up empty libs array
     var libArray = [],
-        libColIsArray  = (libCol.constructor === Array);
+        glob = null;
+
+    function isArray (v) {
+        if (v.constructor === Array) {
```

Comment:
Is the fill `if else` necessary? as it should just stop if constructor isn't an array, thereby returning false? If I'm wrong then we could just make this a shorthand if statement for cleanliness.